### ISSUE TITLE
GlobalProvider type does not have 'children' property

### ISSUE THREAD

**Issue by iamzapata** [tajo/ladle#108]
The [providers section](https://www.ladle.dev/docs/providers) has an example that describes how to add a global provider: 

```
import type { GlobalProvider } from "@ladle/react";

export const Provider: GlobalProvider = ({ children, globalState }) => (
  <>
    <h1>Theme: {globalState.theme}</h1>
    {children}
  </>
);
```

My `components` file in `.ladle/components.tsx` is a typescript file. The children property is not declared in the `GlobalProvider` type, because of this, my IDE gives me a `Property 'children' does not exist on type '{ globalState: GlobalState; }'.` error. 

Here's what the `GlobalProvider` looks like: 

`export type GlobalProvider = React.FC<{ globalState: GlobalState }>;`

Since the example is using TS and it also passes in a children prop, the `GlobalProvider` type should have a children prop.

<img width="910" alt="Screenshot 2022-04-20 at 09 53 08" src="https://user-images.githubusercontent.com/60562158/164178988-67a9a5be-b938-4a8f-be25-b1a7aec755b1.png">

**Comment by tajo** [comment]
I think this happens due to new `@types/react` requiring to define `children` implicitly.

**Comment by johnshift** [comment]
I have this issue when `@types/react` is at `17.0.44`.
Issue was fixed after updating to `17.0.45`.
